### ISSUE TITLE
String.to{Low,Upp}case() gives invalid result

### ISSUE THREAD

**Issue by alexlamsl** [nodejs/node-chakracore#462]
The following fails with `node-chakracore@8.6.0`:
```js
// gives 304 instead
assert.strictEqual("İ".toLowerCase().charCodeAt(0), 105);
// gives 305 instead
assert.strictEqual("ı".toUpperCase().charCodeAt(0), 73);
```
Discovered in https://github.com/mishoo/UglifyJS2/pull/2886#issuecomment-364116274

Other platforms are working as expected:
- IE 8~11
- FireFox 43
- Safari 5
- Opera 10
- Node 0.10~9

**Comment by MSLaguana** [comment]
Looks like we are also inconsistent between different OS: when I try your examples on Linux, the second case works correctly but the first does not. @jackhorton should we be using ICU for upper/lower casing?

**Comment by jackhorton** [comment]
Interesting. This is probably a Chakra bug, rather than a Node-ChakraCore bug, so I will make something to track this over there. With that being said, I dont think ICU should be required here, since the whole point of toLocale{Upper|Lower}Case is that they use locale-sensitive casing rules, so we are probably passing some bad flags to the Win32/POSIX APIs that handle the basic comparison.

**Comment by alexlamsl** [comment]
Still an issue as of `v10.0.0`

**Comment by jackhorton** [comment]
It should only be an issue on Windows. Can you confirm which platform you are running on?

**Comment by alexlamsl** [comment]
@jackhorton sorry for missing this detail in the original report – this is indeed tested on Windows 10 (x64).

**Comment by alexlamsl** [comment]
I can't confirm whether the issue exists on say Ubuntu, as https://github.com/mishoo/UglifyJS2/pull/3087 is blocked by #509, i.e. the process crashed before it can get to the test relevant to this report.

**Comment by alexlamsl** [comment]
Still an issue as of `v10.1.0` (on Windows)

**Comment by alexlamsl** [comment]
Fixed in `v10.6.0` - thanks!
